tools/infer/utility.py

```diff
@@ -692,6 +692,112 @@ def get_minarea_rect_crop(img, points):
     return crop_img
 
 
+def slice_generator(image, horizontal_stride, vertical_stride, maximum_slices=500):
+    if not isinstance(image, np.ndarray):
+        image = np.array(image)
+
+    image_h, image_w = image.shape[:2]
+    vertical_num_slices = (image_h + vertical_stride - 1) // vertical_stride
+    horizontal_num_slices = (image_w + horizontal_stride - 1) // horizontal_stride
+
+    assert (
+        vertical_num_slices > 0
+    ), f"Invalid number ({vertical_num_slices}) of vertical slices"
+
+    assert (
+        horizontal_num_slices > 0
+    ), f"Invalid number ({horizontal_num_slices}) of horizontal slices"
+
+    if vertical_num_slices >= maximum_slices:
+        recommended_vertical_stride = max(1, image_h // maximum_slices) + 1
+        assert (
+            False
+        ), f"Too computationally expensive with {vertical_num_slices} slices, try a higher vertical stride (recommended minimum: {recommended_vertical_stride})"
+
+    if horizontal_num_slices >= maximum_slices:
+        recommended_horizontal_stride = max(1, image_w // maximum_slices) + 1
+        assert (
+            False
+        ), f"Too computationally expensive with {horizontal_num_slices} slices, try a higher horizontal stride (recommended minimum: {recommended_horizontal_stride})"
+
+    for v_slice_idx in range(vertical_num_slices):
+        v_start = max(0, (v_slice_idx * vertical_stride))
+        v_end = min(((v_slice_idx + 1) * vertical_stride), image_h)
+        vertical_slice = image[v_start:v_end, :]
+        for h_slice_idx in range(horizontal_num_slices):
+            h_start = max(0, (h_slice_idx * horizontal_stride))
+            h_end = min(((h_slice_idx + 1) * horizontal_stride), image_w)
+            horizontal_slice = vertical_slice[:, h_start:h_end]
+
+            yield (horizontal_slice, v_start, h_start)
+
+
+def merge_fragmented(boxes, x_threshold=10, y_threshold=10):
+    merged_boxes = []
+    visited = set()
+    merged_counter = 0
+    for i in range(len(boxes)):
+        if i in visited:
+            continue
+
+        current_box = boxes[i]
+        merged_box = [current_box[0], current_box[1], current_box[2], current_box[3]]
+        min_x, max_x, min_y, max_y = (
+            current_box[0][0],
+            current_box[1][0],
+            current_box[0][1],
+            current_box[2][1],
+        )
+
+        for j in range(len(boxes)):
+            if i == j:
+                continue
+
+            compare_box = boxes[j]
+            compare_min_x, compare_max_x, compare_min_y, compare_max_y = (
+                compare_box[0][0],
+                compare_box[1][0],
+                compare_box[0][1],
+                compare_box[2][1],
```

Comment:
Use functions to avoid repetition, the code to calculate `min_x`, `max_x`, `min_y`, and `max_y` is repeated twice. This could be moved into a separate function.